metrics-influxdb/src/main/java/io/dropwizard/metrics5/influxdb/InfluxDbReporter.java

```diff
@@ -0,0 +1,517 @@
+        writeFieldIfEnabled(SUM, histogram.getSum());
+        writeFieldIfEnabled(MAX,    snapshot.getMax());
+        writeFieldIfEnabled(MEAN,   snapshot.getMean());
+        writeFieldIfEnabled(MIN,    snapshot.getMin());
+        writeFieldIfEnabled(STDDEV, snapshot.getStdDev());
+        writeFieldIfEnabled(P50,    snapshot.getMedian());
+        writeFieldIfEnabled(P75,    snapshot.get75thPercentile());
+        writeFieldIfEnabled(P95,    snapshot.get95thPercentile());
+        writeFieldIfEnabled(P98,    snapshot.get98thPercentile());
+        writeFieldIfEnabled(P99,    snapshot.get99thPercentile());
+        writeFieldIfEnabled(P999,   snapshot.get999thPercentile());
+        if (hasValues()) {
+            writeTimestampMillis(timestamp);
+            reportLine();
+        }
+    }
+    
+    
+    private void reportMetered(MetricName name, Metered meter, long timestamp) throws IOException {
+        writeMeasurement(name);
+        writeMeteredFieldsIfEnabled(meter);
+        if (hasValues()) {
+            writeTimestampMillis(timestamp);
+            reportLine();
+        }
+    }
+
+    private void writeMeteredFieldsIfEnabled(Metered meter) {
+        writeFieldIfEnabled(COUNT,     meter.getCount());
+        writeFieldIfEnabled(SUM,       meter.getSum());
+        writeFieldIfEnabled(M1_RATE,   convertRate(meter.getOneMinuteRate()));
+        writeFieldIfEnabled(M5_RATE,   convertRate(meter.getFiveMinuteRate()));
+        writeFieldIfEnabled(M15_RATE,  convertRate(meter.getFifteenMinuteRate()));
+        writeFieldIfEnabled(MEAN_RATE, convertRate(meter.getMeanRate()));
+    }
+    
+    private void reportCounter(MetricName name, Counter counter, long timestamp) throws IOException {
+        writeMeasurement(name);
+        writeFieldIfEnabled(COUNT, counter.getCount());
+        if (hasValues()) {
+            writeTimestampMillis(timestamp);
+            reportLine();
+        }
+    }
+
+    private void reportGauge(MetricName name, Gauge<?> gauge, long timestamp) throws IOException {
+        writeMeasurement(name);
+        Object value = gauge.getValue();
+        if (value != null) {
+            writeField(VALUE);
+            if (value instanceof Number) {
+                Number number = (Number) value;
+                if (number instanceof Long || 
+                        number instanceof Integer || 
+                        number instanceof Short || 
+                        number instanceof Byte) {
+                    writeFieldValue(number.longValue());
+                } else {
+                    writeFieldValue(number.doubleValue());
+                }
+            } else if (value instanceof Boolean) {
+                writeFieldValue(((Boolean)value));
+            } else {
+                writeFieldValue(value.toString());
+            }
+        }
+        writeTimestampMillis(timestamp);
+        reportLine();
+    }    
+    
+    private void reset() {
+        str.delete(0, str.length());
```

Comment:
Looks we could just use `str.setLength(0)` to reset the counter in `StringBuilder`.